src/main/java/delete_post/use_case/DeletePostInteractor.java

```diff
@@ -22,7 +22,7 @@ public void delete(DeletePostRequestModel requestModel){
             this.outputBoundary.prepareTimerView(responseModel);
         }
         else if (CurrentUser.getIsAdmin() ||
-                CurrentUser.getCurrentUser() == post.getUser()){
+                CurrentUser.getCurrentUser().getId() == post.getUser()){
```

Comment:
i think instead we should have CurrentUser.getCurrentUser().equals(post.getUser), as then we're checking equality of the data types

Reply:
Wait nevermind I realized the Post method returns an id